swupd/packs_test.go

```diff
@@ -686,3 +692,188 @@ func checkFileInPack(t *testing.T, packname, name string) {
 	}
 	t.Errorf("file %s is not in pack", name)
 }
+
+func TestPackStateString(t *testing.T) {
+	testCases := []struct {
+		p   PackState
+		exp string
+	}{
+		{NotPacked, "not packed"},
+		{PackedDelta, "packed delta"},
+		{PackedFullfile, "packed fullfile"},
+	}
+
+	for _, tc := range testCases {
+		ps := tc.p.String()
+		if ps != tc.exp {
+			t.Errorf("PackState.String() returned \"%s\"; expected \"%s\"", ps, tc.exp)
+		}
+	}
+
+	p := PackedFullfile
+	p++
+	ps := p.String()
+	if ps != "invalid" {
+		t.Errorf("invalid PackState.String() returned \"%s\"; expected \"invalid\"", ps)
```

Comment:
did you mean to say "valid PackState.String() returned" ?

Reply:
:question:

Reply:
err no, but I'll update it to be a little easier to understand.